libraries/DateTime/contracts/DateTime.aes

```diff
@@ -114,13 +62,191 @@ contract DateTime =
     let timestamp = timestamp + second  
     timestamp
 
-// Private functions and internal logic
+  public function parse_timestamp(timestamp : int) : date_time = 
+    let dt : date_time = init_date_time(timestamp)
+    
+    let dt : date_time = p_get_year(dt)
+    let dt : date_time = p_get_month(dt)
+    let dt : date_time = p_get_day(dt)
+    let dt : date_time = dt{hour = rest(timestamp / 60 /60, 24)}
+    let dt : date_time = dt{minute = rest(timestamp / 60, 60)}
+    let dt : date_time = dt{second = rest(timestamp, 60)}
+    let wd : int = rest(timestamp / state.day_in_seconds + 4, 7)
+    let dt : date_time = dt{weekday = wd}
+    dt
+
+  public function get_year(timestamp : int) : int = 
+    let year : int = parse_timestamp(timestamp).year
+    year
+
+  public function diff_years(from_timestamp : int, to_timestamp : int) : int = 
```

Comment:
would it make sense to define a datatype
```
datatype TimeUnit = Minute | Hour | Day | Month | Year
```
and have one `diff` function:
```
entrypoint diff(time_unit : TimeUnit, from_ts : int, to_ts : int) : int
```
It would save the repetition of the `require` and might even read better?

Reply:
Currently, I am not sure how datatype is handled in javascript in regards to testing such a functionality . Already contacted Naz from #js-sdk. He’ll investigate their side and let us know.

Reply:
whats the result on this?

Reply:
There is open issue already - https://github.com/aeternity/aepp-sdk-js/issues/549